### ISSUE TITLE
Twitter API v2 includes do not update when using async iterators

### ISSUE THREAD

**Issue by akremer** [PLhery/node-twitter-api-v2#142]
I am using async iterators to retrieve a user's timeline with the following code. In the request, I am requesting expansions for each reply the user made.

```
  const tweets = await client.v2.userTimeline('1234567', { 'expansions': ['author_id','in_reply_to_user_id','entities.mentions.username']});

  for await (const fetchedTweet of tweets) {
    console.log(fetchedTweet);
    console.log("-Includes-\n" + JSON.stringify(tweets.includes));
  }
```

According to the [documentation](https://github.com/PLhery/node-twitter-api-v2/blob/master/doc/paginators.md#v2-meta-includes), `tweets.includes` should automatically update based on the available includes collection for each iteration:

```
  // availableMeta and availableIncludes are filled with .meta and .includes
  // fetched at the time you were reading this tweet
  // Once the next page is automatically fetched, they can be updated!
```

This means I should expect the output of tweets.includes to update based on each iteration of the above `for` loop. However, in practice, I only ever receive the `includes` for the first page of 10 results. No matter how much more data the loop fetches, `includes` is never updated and contains the exact same result every time.

If I change the code to use `fetchLast()`, it works perfectly:

```
 const tweets = await client.v2.userTimeline('1234567', { 'expansions': ['author_id','in_reply_to_user_id','entities.mentions.username']});

  await tweets.fetchLast(1000);

  for (const fetchedTweet of tweets) {
    console.log(fetchedTweet);
    console.log("-Includes-\n" + JSON.stringify(tweets.includes));
  }
```

Using the above code, `tweets.includes` correctly displays the `includes` for every result in the result set.

It seems that using async iterators, includes is somehow not being updated properly. Could you take a look? Thanks.

**Comment by alkihis** [comment]
Hello,

You're right, `includes` and `meta` *should* be updated according to the documentation.
Unfortunately, `includes` are stored into current paginator instance, and async iterators do **not** mutate the current instance.
Implementing a fix for this will introduce a breaking change for users relying of immutability induced by async iterator.

Three solutions:
- Make paginators store into current instance fetched results during async iteration
- Make paginators store **only** metadata into current instance during async iteration, but this introduce a weird behavior where data for things that aren't stored is available into the instance
- Implement a new async generator, like `.fetchAndIterate()`, that yields a tuple `[item, currentPaginator]` for each fetched item, giving the availability to access metadata:
```ts
const tweets = await client.v2.userTimeline('1234567', { 'expansions': ['author_id','in_reply_to_user_id','entities.mentions.username'] });

for await (const [fetchedTweet, paginator] of tweets.fetchAndIterate()) {
  console.log(fetchedTweet);
  console.log("-Available includes-\n" + JSON.stringify(paginator.includes));
}
```

**Comment by akremer** [comment]
Thanks for confirming the issue. In my very humble opinion, the third seems the cleanest and easy to understand and document.

**Comment by alkihis** [comment]
Update: Fix is available in 1.9.0 :)
